engine/src/main/java/io/camunda/zeebe/engine/processing/bpmn/behavior/LegacyTypedStreamWriterProxy.java

```diff
@@ -8,17 +8,17 @@
 package io.camunda.zeebe.engine.processing.bpmn.behavior;
 
 import io.camunda.zeebe.engine.api.TypedRecord;
-import io.camunda.zeebe.engine.processing.streamprocessor.writers.TypedStreamWriter;
+import io.camunda.zeebe.engine.processing.streamprocessor.writers.LegacyTypedStreamWriter;
 import io.camunda.zeebe.protocol.record.RecordType;
 import io.camunda.zeebe.protocol.record.RecordValue;
 import io.camunda.zeebe.protocol.record.RejectionType;
 import io.camunda.zeebe.protocol.record.intent.Intent;
 
-public final class TypedStreamWriterProxy implements TypedStreamWriter {
+public final class LegacyTypedStreamWriterProxy implements LegacyTypedStreamWriter {
```

Comment:
❓ Maybe I am misunderstanding the Legacy stuff. My thoughts were that these were going to be removed entirely. They are still here, do we need to keep these or will they be removed later in the abstraction topic?

Reply:
I could mark them as deprecated, but CodeQL took the fun out of it. It will report a use of a deprecated method as an issue and then you manually have to confirm that it was intended.

Eventually, yes, they will be removed. But there are more steps to take in the engine abstraction topic. To describe it in a metaphor, we have built a new bridge for traffic go across, but the old ways are still used in some places. 

For example `LegacyTypedCommandWriter` is still used in `JobTimeoutTrigger`, which implements a background task to deactivate timed out tasks. It is classes like this, which are adjacent to the engine, but not part of the engine in terms of record processing which still use the old code.

Continuation of the engine abstraction will be to refactor all these places and move them also over to the next code. But this might take a while. Consider for example #8002 which has been open for some time now.